### PR TITLE
[cli] Fix prebuild path resolving on Windows

### LINKED ISSUE #31403: Android prebuild issue with SDK 52 canary
### Summary

Hey!

I initialized an app using `npx create-expo-app` and then upgraded it to use the latest canary release using `npm install expo@canary` since I wanted to expirement with the new DOM components feature. 

Now, I changed nothing in the code (so it's just the basic App.js saying 'Open up App.js to start working on your app!'). When I ran `npx expo run:android` and completed all steps in the cli, it starts by creating the android folder. When it starts prebuilding, it crashes immediately and shows the error:
```
AssertionError [ERR_ASSERTION]: [android.dangerous]: withAndroidDangerousBaseMod: Project file "MainApplication" does not exist in android project for root "[APP PATH]"
```

When I go check if the file exists, I can see there is a `MainApplication.kt` file in the android folder.

This only happens with the canary since doing the exact same thing with SDK 51 does work and build the app as expected.

Are there any solutions for this?

--------------------------------------------------------------------------------------
The exact versions that my app uses are:
```
"dependencies": {
    "expo": "^52.0.0-canary-20240904-69100c1",
    "expo-status-bar": "1.12.2-canary-20240904-69100c1",
    "react": "18.3.1",
    "react-native": "0.75.2",
    "react-native-webview": "13.8.6"
  },
```

### What platform(s) does this occur on?

Android

### SDK Version

52.0.0-canary-20240904-69100c1

### Environment

expo-env-info 1.2.0 environment info:
    System:
      OS: Windows 11 10.0.22631
    Binaries:
      Node: 20.12.2 - D:\Programs\NodeJS\node.EXE
      Yarn: 1.22.21 - ~\AppData\Roaming\npm\yarn.CMD
      npm: 10.5.0 - D:\Programs\NodeJS\npm.CMD
    npmPackages:
      expo: ^52.0.0-canary-20240904-69100c1 => 52.0.0-canary-20240904-69100c1
      react: 18.3.1 => 18.3.1
      react-native: 0.75.2 => 0.75.2
    Expo Workflow: bare

### Minimal reproducible example

https://github.com/StefSchalks/expo-android-run-test

### PR DESCRIPTION
**Pull request by LeonDvlpmnt**
# Why

Fixes https://github.com/expo/expo/issues/31403. This looks to be a regression from updating to `glob@10` (cc @byCedric )

# How

`globSync` expects unix style paths. `path.join` returns windows style paths. With this change it will always pass unix style paths to glob. 

node-glob also has a `windowsPathsNoEscape` option but they discourage using it. That's why I opted for the regex option.

# Test Plan

A bit difficult to test. You need to test it on Windows without WSL.  You can verify this works by running `prebuild` with and without this change. Without this change `prebuild` will fail.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by leonhh** [comment]
Oh, this is annoying. The add `addMapsCocoaPods` test is failing because of changes to the Podfile of the bare template (for the new autolinking). That's a bit out of scope for this PR.

**Comment by leonhh** [comment]
https://github.com/expo/expo/pull/32282 should fix the failing test

File: packages/@expo/config-plugins/src/android/Paths.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ export type GradleProjectFile = ProjectFile<'groovy' | 'kt'>;
 
 export function getProjectFilePath(projectRoot: string, name: string): string {
   const filePath = globSync(
```
Comment (EvanBacon) [inline packages/@expo/config-plugins/src/android/Paths.ts:20]: Maybe use the `{ posix: true }` option in glob?

Comment (byCedric) [reply packages/@expo/config-plugins/src/android/Paths.ts:20]: We can also use `path.posix.join`

Comment (leonhh) [reply packages/@expo/config-plugins/src/android/Paths.ts:20]: This uses both the regex and path.posix.join:

https://github.com/expo/expo/blob/4f7c4948c736de5042ba99c353a46e92668bf0c3/packages/%40expo/metro-config/src/getWatchFolders.ts#L35 

I would assume only one of these is needed. I will test this on my Windows PC later today.

**Pull request closed without merging**